From 576b831ae19ceba05701171efbb03febde37757f Mon Sep 17 00:00:00 2001
From: Meng Dongyang <daniel.meng@rock-chips.com>
Date: Mon, 11 Jun 2018 17:08:44 +0800
Subject: [PATCH] usb: gadget: uac: correct f_audio_source.c to remove warning

If the CONFIG_USB_CONFIGFS is not define, the 'audio_source_bind_config'
may be unused. Corrcet warning of no previous prototype for
'audio_source_bind_config' by adding judgment of CONFIG_USB_CONFIGFS
macro definition.

Change-Id: I174ae996c1d53e78b76f72cf463bc1efb189675b
Signed-off-by: Meng Dongyang <daniel.meng@rock-chips.com>
---
 drivers/usb/gadget/function/f_audio_source.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/gadget/function/f_audio_source.c b/drivers/usb/gadget/function/f_audio_source.c
index a9a725101026..bb7e496bfdac 100644
--- a/drivers/usb/gadget/function/f_audio_source.c
+++ b/drivers/usb/gadget/function/f_audio_source.c
@@ -1215,6 +1215,7 @@ static struct snd_pcm_ops audio_playback_ops = {
 	.pointer	= audio_pcm_pointer,
 };
 
+#ifndef CONFIG_USB_CONFIGFS
 int audio_source_bind_config(struct usb_configuration *c,
 		struct audio_source_config *config)
 {
@@ -1253,6 +1254,7 @@ int audio_source_bind_config(struct usb_configuration *c,
 	snd_card_free(audio->card);
 	return err;
 }
+#endif
 
 static int snd_card_setup(struct usb_configuration *c,
 		struct audio_source_config *config)
-- 
2.35.3

